src/core/public/application/types.ts

```diff
@@ -66,8 +81,55 @@ export interface AppBase {
    * Custom capabilities defined by the app.
    */
   capabilities?: Partial<Capabilities>;
+
+  /**
+   * Flag to keep track of legacy applications.
+   * For internal use only. any value will be overridden when registering an App.
+   *
+   * @internal
+   */
+  legacy?: boolean;
+
+  /**
+   * Hide the UI chrome when the application is mounted. Defaults to `false`.
+   * Takes precedence over chrome service visibility settings.
+   */
+  chromeless?: boolean;
+}
+
+/**
+ * Accessibility status of an application.
+ *
+ * @public
+ */
+export enum AppStatus {
+  /**
+   * Application is accessible.
+   */
+  accessible = 0,
+  /**
+   * Application is not accessible, but visible in an inactive state in navigation.
+   */
+  inaccessibleWithDisabledNavLink = 1,
+  /**
+   * Application is not accessible.
+   */
+  inaccessible = 2,
```

Comment:
I think we have one more case that is not covered here: apps that are accessible but have no icon in the nav. For example, the /status page has a custom route, has no navlink, but does show the chrome (so it's not 'chromeless' either).

Reply:
Good point.

This makes me think that *maybe* the initial proposition to have distinct status for navlink and application may makes sense... But adding it to the enum should work too.